app/src/main/java/com/jalasoft/routesapp/ui/auth/login/view/LoginFragment.kt

```diff
@@ -1,32 +1,166 @@
+        observers()
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        FirebaseApp.initializeApp(context)
     }
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        return inflater.inflate(R.layout.fragment_login, container, false)
+    ): View {
+        _binding = FragmentLoginBinding.inflate(inflater, container, false)
+        return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        view.findViewById<Button>(R.id.btn_log_register).setOnClickListener {
+        googleConfiguration()
+        viewModel.context = context
+        binding.btnLogRegister.setOnClickListener {
             findNavController().navigate(R.id.action_loginFragment_to_registerUserFragment)
         }
+
+        binding.btnLogin.setOnClickListener {
+            loginWithEmailAndPassword()
+        }
+
+        binding.ibLogGoogle.setOnClickListener {
+            signInGoogle()
+        }
+    }
+
+    private fun googleConfiguration() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_cliente_id))
+            .requestEmail()
+            .build()
+        googleSingInClient = GoogleSignIn.getClient(binding.root.context, gso)
+    }
+
+    private fun loginWithEmailAndPassword() {
+        showProgress(true)
+        val email = binding.loginEmail.text.toString()
+        val password = binding.loginPassword.text.toString()
+
+        viewModel.loginUserAuth(email, password)
+    }
+
+    private fun observers() {
+        val errorObserver = Observer<String> { errorMessage ->
+            binding.txtError.isVisible = true
+            binding.txtError.text = errorMessage
+            showProgress(false)
+        }
+        val resultObserver = Observer<Boolean> {
+            showProgress(false)
+            binding.loginEmail.setText("")
+            binding.loginPassword.setText("")
+            findNavController().navigate(R.id.homeFragment)
+            Toast.makeText(context, "Login Successfully", Toast.LENGTH_SHORT).show()
+        }
+        val googleObserver = Observer<Boolean> { value ->
+            if (value) {
+                findNavController().navigate(R.id.homeFragment)
+                showProgress(false)
+            }
+        }
+        viewModel.errorMessage.observe(this, errorObserver)
+        viewModel.loginIsSuccessful.observe(this, resultObserver)
+        viewModel.signInGoogle.observe(this, googleObserver)
+    }
+
+    private fun showProgress(show: Boolean) {
+        if (show) {
+            binding.pbLogin.visibility = View.VISIBLE
+        } else {
+            binding.pbLogin.visibility = View.GONE
+        }
+    }
+
+    /*fun goToHomeFragment(){
```

Comment:
Delete commented lines

Reply:
This is from Adriana's branch she didn't finish it